### DISCUSSION TITLE
How janusgraph cluster traversal gtraph ? How the traversal task split to the cluster node ?

### DISCUSSION THREAD

**Discussion by TouSS** [JanusGraph/janusgraph#2905] [Q&A] [answered]
I don't understand how janusgraph cluster worked when traversal gtraph：
* Only one of the cluster node choosed to traversal the traversal ?
* OR the traversal can be split to some of the cluster nodes, If so, how it work?

Anyone can tell how please? 

Thanks and regards

**Comment by li-boxuan** [comment, marked as answer]
Only one node will be used to run the traversal. If you have a huge traversal that takes too long or too much memory on a single node, you could leverage [SparkGraphComputer](https://docs.janusgraph.org/advanced-topics/hadoop/) to run the traversal in a distributed manner.

**Reply by TouSS** [reply]
Got it.
